Source/Bugsnag/BugsnagNotifier.m

```diff
@@ -114,27 +148,29 @@ - (void)notify:(NSException *)exception withData:(NSDictionary *)metaData atSeve
     }
 
     [self.metaDataLock lock];
-    [self serializeDictionary: metaData toJSON: &g_bugsnag_data.metaDataJSON];
-    [self.state addAttribute:@"severity" withValue: severity toTabWithName: @"crash"];
-    [self.state addAttribute:@"depth" withValue: [NSNumber numberWithUnsignedInteger:depth + 3] toTabWithName: @"crash"];
-    [self serializeBreadcrumbs];
-    NSString *exceptionName = [exception name] != nil ? [exception name] : @"NSException";
+    BSSerializeJSONDictionary(metaData, &g_bugsnag_data.metaDataJSON);
+    [self.state addAttribute:BSAttributeSeverity withValue:severity toTabWithName:BSTabCrash];
+    [self.state addAttribute:BSAttributeDepth withValue:@(depth + 3) toTabWithName:BSTabCrash];
+    NSString *exceptionName = [exception name] ?: NSStringFromClass([NSException class]);
     [[KSCrash sharedInstance] reportUserException:exceptionName reason:[exception reason] lineOfCode:@"" stackTrace:@[] terminateProgram:NO];
 
     // Restore metaData to pre-crash state.
     [self.metaDataLock unlock];
     [self metaDataChanged: self.configuration.metaData];
-    [[self state] clearTab:@"crash"];
+    [[self state] clearTab:BSTabCrash];
 
     [self performSelectorInBackground:@selector(sendPendingReports) withObject:nil];
+    [self resetDefaultSeverity];
+}
+
+- (void) resetDefaultSeverity {
```

Comment:
I dont think so. I think we'd just be saying that if one wasnt written it was in fact an error. I dont mind which direction we head in but wanted to point it out as an option.